libs/platform/src/lib/components/form/checkbox-group/checkbox-group.component.spec.ts

```diff
@@ -0,0 +1,451 @@
+                    [formControl]="fl1.formControl"
+                ></fdp-checkbox-group>
+            </fdp-form-field>
+
+            <fdp-form-field #fl2 [id]="'visited'" [label]="'Country visited: '" zone="zLeft" rank="1">
+                <fdp-checkbox-group
+                    [list]="countryVisited"
+                    [name]="'visited'"
+                    [formControl]="fl2.formControl"
+                ></fdp-checkbox-group>
+            </fdp-form-field>
+
+            <fdp-form-field #fl3 [id]="'hobbies'" [label]="'My Hobbies:'" zone="zLeft" rank="1">
+                <fdp-checkbox-group [name]="'hobby'" [formControl]="fl3.formControl">
+                    <fdp-checkbox [value]="'cooking'" [label]="'Cooking'"></fdp-checkbox>
+                    <fdp-checkbox [value]="'painting'" [label]="'Painting'"></fdp-checkbox>
+                    <fdp-checkbox [value]="'coding'" [label]="'Coding'"></fdp-checkbox>
+                    <fdp-checkbox [value]="'gardening'" [label]="'Gardening'"></fdp-checkbox>
+                </fdp-checkbox-group>
+            </fdp-form-field>
+
+            <fdp-form-field #fl4 [id]="'language'" [label]="'Languages Known: '" zone="zLeft" rank="1">
+                <fdp-checkbox-group
+                    [list]="languages"
+                    [name]="'language'"
+                    [formControl]="fl4.formControl"
+                ></fdp-checkbox-group>
+            </fdp-form-field>
+
+            <fdp-form-field #fl5 [id]="'fruits'" [label]="'Fruits:'" zone="zLeft" rank="1">
+                <fdp-checkbox-group [name]="'fruits'" [formControl]="fl5.formControl">
+                    <fdp-checkbox [value]="'apple'" [label]="'Apple'" [disabled]="true"></fdp-checkbox>
+                    <fdp-checkbox [value]="'banana'" [label]="'Banana'"></fdp-checkbox>
+                    <fdp-checkbox [value]="'guava'" [label]="'Guava'" [disabled]="true"></fdp-checkbox>
+                    <fdp-checkbox [value]="'papaya'" [label]="'Papaya'"></fdp-checkbox>
+                </fdp-checkbox-group>
+            </fdp-form-field>
+        </fdp-form-group>
+    `
+})
+class TestReactiveCheckboxGroupComponnet {
+    phoneslist: string[] = ['Samsung', 'Apple', 'OnePlus', 'Redmi'];
+    countryVisited = [new Country('Australia', 'Australia'), new Country('India', 'India'), new Country('USA', 'USA')];
+
+    languages = [
+        new LanguageKnown('Java', 'java', false),
+        new LanguageKnown('Javascript', 'javascript', true),
+        new LanguageKnown('Python', 'python', false),
+        new LanguageKnown('GoLang', 'go', true)
+    ];
+
+    form1 = new FormGroup({});
+    formData = { phones: ['Samsung', 'OnePlus'], visited: ['India', 'USA'], hobbies: ['coding', 'gardening'] };
+
+    @ViewChildren(CheckboxComponent)
+    checkboxGroups: QueryList<CheckboxGroupComponent>;
+}
+
+describe('CheckboxGroup component Reactive Form Test', () => {
+    let host: TestReactiveCheckboxGroupComponnet;
```

Comment:
You want to also do more reactive tests where you initialize formGroup with values, I see you have object which is good. but FormGroup is needed as well.

Reply:
included tests and examples for formGroup with values.